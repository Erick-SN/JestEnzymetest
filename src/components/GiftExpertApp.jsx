import React, { useState } from 'react';
import AddCategory from './AddCategory';
import GiftGrid from './GiftGrid';
const GiftExpertApp = ({ defaultCategory = [] }) => {
  const [categories, setCategories] = useState(defaultCategory);

  return (
    <>
      <h2>My app</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => (
          <GiftGrid key={category} category={category} />
        ))}
      </ol>
    </>
  );
};

export default GiftExpertApp;
