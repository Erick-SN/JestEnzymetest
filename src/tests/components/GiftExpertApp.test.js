import React from 'react';
import { shallow } from 'enzyme';
import '@testing-library/jest-dom';
import GiftExpertApp from '../../components/GiftExpertApp';

describe('Test at <GiftExpertApp />', () => {
  test('Render <GiftExpertApp />', () => {
    const wrapper = shallow(<GiftExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('Should show a list of categories', () => {
    const categories = ['One Punch', 'Dragon Ball'];
    const wrapper = shallow(<GiftExpertApp defaultCategory={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GiftGrid').length).toBe(categories.length);
  });
});
