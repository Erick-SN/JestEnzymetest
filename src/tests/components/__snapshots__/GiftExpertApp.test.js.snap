// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <GiftExpertApp /> Render <GiftExpertApp /> 1`] = `ShallowWrapper {}`;

exports[`Test at <GiftExpertApp /> Should show a list of categories 1`] = `ShallowWrapper {}`;
